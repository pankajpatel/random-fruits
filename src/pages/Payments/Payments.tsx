import { FormattedDate, FormattedMessage } from 'react-intl';
import { useLocation } from 'wouter';

import { Badge } from '@app/ds/Badge';
import { Spinner } from '@app/ds/Spinner';
import { Table, TableCell } from '@app/ds/Table/Table';
import { SORT_DIRECTIONS } from '@app/constants/sorting';
import { usePayments } from '@app/hooks/usePayments';
import { getBadgeColorForStatus } from '@app/utils/getBadgeColorForStatus';
import { PageHeader } from '@app/components/PageHeader/PageHeader';
import { Amount } from '@app/components/Amount/Amount';
import { FilterBar } from '@app/components/FilterBar/FilterBar';

const CellConfig: Array<TableCell<PaymentInList>> = [
  {
    label: (
      <FormattedMessage
        id="table.header.customer"
        defaultMessage="Customer Name"
      />
    ),
    key: 'customer_name',
    maxWidth: '30%',
  },
  {
    label: (
      <FormattedMessage id="table.header.merchant" defaultMessage="Merchant" />
    ),
    key: 'merchant',
    render: (row: PaymentInList) => row.merchant.name,
    maxWidth: '30%',
  },
  {
    label: (
      <FormattedMessage id="table.header.amount" defaultMessage="Amount (â‚¬)" />
    ),
    key: 'amount',
    align: 'right',
    sortable: true,
    render: (row: PaymentInList) => <Amount value={row.amount} />,
  },
  {
    label: <FormattedMessage id="table.header.created" defaultMessage="On" />,
    key: 'created',
    render: (row: PaymentInList) => (
      <FormattedDate value={new Date(row.created)} dateStyle="long" />
    ),
  },
  {
    label: (
      <FormattedMessage id="table.header.status" defaultMessage="Status" />
    ),
    key: 'status',
    render: (row: PaymentInList) => (
      <Badge type={getBadgeColorForStatus(row.status)}>
        <FormattedMessage
          id={`status.${row.status}`}
          defaultMessage={row.status}
        />
      </Badge>
    ),
    sortable: true,
    align: 'right',
  },
];

export const Payments = (): JSX.Element => {
  const { payments, isError, isLoading, sortRows, filter } = usePayments();

  const [, gotTo] = useLocation();

  return (
    <section>
      <PageHeader>
        <FormattedMessage id="page.payments.title" defaultMessage="Payments" />
      </PageHeader>
      <FilterBar onChange={filter} />
      {isLoading && <Spinner />}
      {isError && <div>Failed to load</div>}
      {!isLoading && !isError && payments ? (
        <Table
          rows={payments}
          cells={CellConfig}
          onRowClick={(row: PaymentInList) => gotTo(`/payments/${row.id}`)}
          onHeaderClick={(key: keyof PaymentInList) => {
            sortRows(key as keyof PaymentInList, SORT_DIRECTIONS.ASC);
          }}
          noResultsMessage={
            <FormattedMessage
              id="page.payments.table.no_data"
              defaultMessage="No data"
            />
          }
        />
      ) : null}
    </section>
  );
};

export default Payments;
