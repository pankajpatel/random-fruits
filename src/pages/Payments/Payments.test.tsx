import { render } from "@testing-library/react";
import { Payments } from './Payments';

import { withProviders } from "@app/testing/provider-setup";

describe('Payments', () => {
  it('should render when no transaction is present', () => {
    const { container } = render(withProviders(<Payments />));
    expect(container).toMatchSnapshot();
  })

  it('should render when transactions are present', () => {
    const { container } = render(withProviders(<Payments />));
    expect(container).toMatchSnapshot();
  })
})
