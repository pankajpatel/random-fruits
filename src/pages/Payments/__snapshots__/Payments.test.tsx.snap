// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payments should render when no transaction is present 1`] = `
.c4 {
  border: 2px solid transparent;
  border-left-color: #333;
  border-radius: 50%;
  width: 1.2rem;
  height: 1.2rem;
  display: block;
  margin: 0.25rem;
  -webkit-animation: keep-spinning 500ms infinite;
  animation: keep-spinning 500ms infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 0 0.5rem;
  margin-bottom: 1.5rem;
}

.c2 {
  font-size: 1.2rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.25rem;
  width: 100%;
}

.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c3 {
  min-width: 300px;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.35em;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5;
  margin: 0.5rem;
}

<div>
  <section>
    <header
      class="c0"
    >
      <h2
        class="c1"
      >
        Payments
      </h2>
      <div
        class=""
      >
        <select
          class="c2"
        >
          <option
            value="en"
          >
            ðŸ‡¬ðŸ‡§
          </option>
          <option
            value="fr"
          >
            ðŸ‡«ðŸ‡·
          </option>
        </select>
      </div>
    </header>
    <div>
      <input
        class="c3"
        placeholder="Filter Payments"
        type="text"
      />
    </div>
    <div
      class="c4"
    />
  </section>
</div>
`;

exports[`Payments should render when transactions are present 1`] = `
.c4 {
  border: 2px solid transparent;
  border-left-color: #333;
  border-radius: 50%;
  width: 1.2rem;
  height: 1.2rem;
  display: block;
  margin: 0.25rem;
  -webkit-animation: keep-spinning 500ms infinite;
  animation: keep-spinning 500ms infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 0 0.5rem;
  margin-bottom: 1.5rem;
}

.c2 {
  font-size: 1.2rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.25rem;
  width: 100%;
}

.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c3 {
  min-width: 300px;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.35em;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5;
  margin: 0.5rem;
}

<div>
  <section>
    <header
      class="c0"
    >
      <h2
        class="c1"
      >
        Payments
      </h2>
      <div
        class=""
      >
        <select
          class="c2"
        >
          <option
            value="en"
          >
            ðŸ‡¬ðŸ‡§
          </option>
          <option
            value="fr"
          >
            ðŸ‡«ðŸ‡·
          </option>
        </select>
      </div>
    </header>
    <div>
      <input
        class="c3"
        placeholder="Filter Payments"
        type="text"
      />
    </div>
    <div
      class="c4"
    />
  </section>
</div>
`;
